qiskit_nature/operators/second_quantization/second_quantized_op.py

```diff
@@ -54,3 +54,12 @@ def reduce(self, atol: Optional[float] = None, rtol: Optional[float] = None):
             The reduced operator`
         """
         raise NotImplementedError
+
+    @abstractmethod
+    def to_list(self) -> List[Tuple[str, complex]]:
+        """Returns the operators internal contents in list-format."""
+        raise NotImplementedError
+
+    def is_hermitian(self) -> bool:
+        """Checks whether the operator is hermitian"""
+        return frozenset(self.to_list()) == frozenset(self.adjoint().to_list())
```

Comment:
In creating a set, from a list, any duplicates in the list would get dropped of course. So as long as the adjoint operator has the exact same number terms, and nothing is simplified, the corresponding duplicates, would get dropped there too. Hence given the current code, I think this technique works. Maybe tests should have some more complex examples that would cover this aspect

Reply:
Yes that is the idea behind this method as a workaround of not having an exact `__eq__` method for the `SecondQuantizedOps`. When there is a `to_matrix`-method in place for all operator types this could be used as a more precise (albeit expensive) method of comparison.

Reply:
Applying `reduce()` before the equality check may be nice to remove duplicated terms.But as @LaurinFischer said, this is workaround until `to_matrix()`. It might be nice to merge this PR after other PRs including `to_matrix()`.

Reply:
to_matrix() though seems like an expensive solution to this. My comment was also to point out that if anyone change adjoint() such that the resultant operator was simplified then the list based test would fail I think - as such in the unit tests it would be good to have one that has duplicates that we might catch such a change (We have had problems with list/set based comparison in the past hence my commenting etc)

Reply:
Thanks for the feedback. I agree with Steve and have made sure that the tests are performed on operators that do have duplicate terms in their `to_list()` such that if the implementations of `.adjoint()` were to change this should be caught.

Reply:
Nice, thanks. But since each subclass has their own adjoint implementation might it be prudent to do a unit test for each, i.e. include SpinOp and VibrationalOp to assure things work as expected.

Reply:
As Ikko suggested above, I added a `reduce()` before the comparison to further proof against any duplicate terms being wrongfully discarded. Also tests now include all operator types, i.e. FermionicOp, SpinOp and VibrationalOp.